sdk/python/lib/pulumi/resource.py

```diff
@@ -13,88 +13,90 @@
 # limitations under the License.
 
 """The Resource module, containing all resource-related definitions."""
-from __future__ import absolute_import
-import six
+from typing import Optional, List, Any, TYPE_CHECKING
 
 from .runtime import known_types
 from .runtime.resource import register_resource, register_resource_outputs
 from .runtime.settings import get_root_resource
-from .runtime.unknown import Unknown
 
-class Resource(object):
+if TYPE_CHECKING:
```

Comment:
I'm not sure what this does. Is it worth commenting?

Reply:
Sure: https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles